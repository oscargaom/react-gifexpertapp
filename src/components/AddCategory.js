import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('')

    const handleChange = (e) => {
        setInputValue(e.target.value);
        //console.log(`handleChange take place`);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        //console.log(`handleSubmit take place with value: "${inputValue}"`);

        if (inputValue.trim().length > 0 ) {
            setCategories(cat => [inputValue, ...cat]);
            //console.log(`setCategories called with value: "${inputValue}"`);
            setInputValue('');
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
