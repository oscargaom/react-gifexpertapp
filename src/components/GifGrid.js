// import React, { useState, useEffect } from 'react'
import React from 'react'
import { useFecthGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';
// import { getGifs } from '../helpers/GetGifs';

export const GifGrid = ({category}) => {

    // const [images, setImages] = useState([]);

    // getGifs();
    const {data: images, loading} = useFecthGifs(category);
    // console.log(data);
    // console.log(loading);


    return (
        <>
            <h3 className="animate__animated animate__fadein">{category}</h3>
            {
                loading && <p className="animate__animated animate__flash">Loading...</p>
            }
            <div className="card-grid">
                {
                    images.map( img => (
                        <GifGridItem 
                            key={img.id} 
                            {...img} 
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}