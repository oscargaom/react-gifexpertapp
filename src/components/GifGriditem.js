import React from 'react'

export const GifGriditem = ({title, url}) => {


    return (
        <div className="card animate__animated animate__fadein">
            <img src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}
