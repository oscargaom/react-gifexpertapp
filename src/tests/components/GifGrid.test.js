import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid';
import { useFecthGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas unitarias del componente <GifGrid />', () => {
    const category = 'Homero';

    test('Validar que el componente se cargue correctamente', () => {

        useFecthGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar imagenes al hacer el llamado al api', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://sitio.com/imagen.jpg',
            title: 'Homero bailarin'
        },
        {
            id: '123',
            url: 'https://sitio.com/imagen.jpg',
            title: 'Homero bailarin'
        }];

        useFecthGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false);
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
    })
    
        
})
