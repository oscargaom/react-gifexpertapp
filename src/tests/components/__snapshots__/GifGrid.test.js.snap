// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas unitarias del componente <GifGrid /> Debe mostrar imagenes al hacer el llamado al api 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadein"
  >
    Homero
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Homero bailarin"
      url="https://sitio.com/imagen.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Homero bailarin"
      url="https://sitio.com/imagen.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas unitarias del componente <GifGrid /> Validar que el componente se cargue correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadein"
  >
    Homero
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
